test/esinstall/node-env/node-env.test.js

```diff
@@ -1,35 +1,56 @@
+const fs = require('fs');
 const path = require('path');
-const {
-  existsPackageJson,
-  runTest,
-  testLockFile,
-  testWebModules,
-} = require('../esinstall-test-utils.js');
-
-require('jest-specific-snapshot'); // allows to call expect().toMatchSpecificSnapshot(filename, snapshotName)
+const {install} = require('../../../esinstall/lib');
 
 describe('node-env', () => {
-  it('matches the snapshot', async () => {
-    const cwd = __dirname;
-
-    if (existsPackageJson(cwd) === false) return;
-
-    // Run Test
-    const {output, snapshotFile} = await runTest(cwd);
+  it('inlines data if specified', async () => {
+    const dest = path.join(__dirname, 'test-basic');
+
+    const env = {
+      ENV_STRING: 'string',
+      ENV_NUMBER: 23,
+      ENV_BOOLEAN: true,
+      ENV_ARRAY: [1],
+      ENV_OBJECT: {obj: true},
+      ENV_NULL: null,
+      ENV_UNDEFINED: undefined,
+    };
+
+    await install(['node-env-mock-pkg'], {dest, env, cwd: 'test-success'});
+
+    const nodeEnvMockPkg = fs.readFileSync(path.join(dest, 'node-env-mock-pkg.js'), 'utf8');
+
+    // positive test
+    expect(nodeEnvMockPkg).toEqual(expect.stringContaining(`const string = "string";`));
```

Comment:
@matthewp just a note: you don’t have to copy this test pattern if you don‘t like it; I just use it a) because `expect.stringContaining()` prints out a clear diff on mismatch (if you just use `expect(str.includes('…').toBe(true)` it will just fail w/o showing you the diff.). And b) because with individual lines you‘re not having to deal with Windows line breaks and you‘re also saying the order of these doesn‘t matter.

But it can be a bit noisy, and may not work in all cases, so use what makes sense to you. I’m just yammering about what I‘ve found to be helpful.